### ISSUE TITLE
Unexpected error using balanced option in spaced-comment rule

### ISSUE THREAD

**Issue by meeber** [eslint/eslint#6689]
**What version of ESLint are you using?**

3.1.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```yaml
rules:
  spaced-comment:
    - error
    - always
    - block:
        balanced: true
```

**What did you do? Please include the actual source code causing the issue.**

Linted this file:

```js
/* this line is fine */

/*
 * these lines are fine
 */

// this line is reported as a problem
```

**What did you expect to happen?**

- No linting errors to be detected.
- The `balanced` option to only be considered when analyzing `/* */` style comments, not `//` style.

**What actually happened? Please include the actual, raw output from ESLint.**

Encountered this linting error:

```
7:1  error  Expected space or tab before '*/' in comment  spaced-comment

✖ 1 problem (1 error, 0 warnings)
```

**Comment by ilyavolodin** [comment]
Confirmed. Although I'm not 100% sure if the rule is failing on the inline comment, or at the end of the jsdoc style comment. Location seems to be pointing at inline comment, but content seems to indicate that it's jsdoc comment that's causing an issue.

**Comment by meeber** [comment]
Looks like changing the file so it only contains one line results in the same error:

File:

```js
// this line is reported as a problem
```

Result:

```
  1:1  error  Expected space or tab before '*/' in comment  spaced-comment

✖ 1 problem (1 error, 0 warnings)

```
